### DISCUSSION TITLE
Both flashing and mass storage on UART - possible??

### DISCUSSION THREAD

**Discussion by ishahak** [adafruit/tinyuf2#314] [Q&A] [answered]
When using the Espressif ESP32S3 DevkitC, the behavior is that I can flash the board using the UART connector, and connect as mass storage on the USB connector.

Is it possible to make a change to the build, so that **both functionalities are done for the UART connector**?

It is important to me because I need to adapt TinyUF2 to a custom board which has only UART with CP2102 hardware.

I spent many hours trying to figure this out, without success. Hoping you can help!
thanks!

**Comment by hathach** [comment]
it is not possible.

**Reply by ishahak** [reply]
Hi, thanks for the response and thanks for this amazing project which I enjoy very much!
I am wondering how it is working for a board like [ESP32-S2 Saola-1](https://circuitpython.org/board/espressif_saola_1_wroom/), for which the [schematics](https://dl.espressif.com/dl/schematics/ESP32-S2-SAOLA-1_V1.1_schematics.pdf) show clearly that the USB is connected to the UART?

**Reply by hathach** [reply, marked as answer]
you will need 2 jumper wires and usb connector breakout to use with saola board

**Reply by ishahak** [reply]
Ok thank you! Too bad we have already built our board with the wrong component...
